### PR TITLE
Group becomes ACTIVE in beginning of convergence cycle

### LINKED ISSUE #1809: ERROR Group should become ACTIVE at beginning of convergence
When user finds ERROR group with information about invalid launch config, he/she will fix the launch config and call `converge`. At that time, the group status is still shown as ERROR and is updated to ACTIVE only after convergence succeeds which can be multiple cycles resulting in a update potentially after 1-2 minutes depending on the group and number of resources created/deleted. During that time the group is technically _active_ and working to bring desired state but shows ERROR. This is incorrect and results in user assuming that `converge` didn't work. The group should move to ACTIVE at beginning of convergence cycle. This way it is accurate with its definition.

### PR DESCRIPTION
**Pull request by manishtomar**
Closes #1809

### CODE REVIEW

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-3f7004b413/docs/autoscale/v1/developer-guide/). :bow:

**Comment by manishtomar** [comment]
jenkins please run integration tests

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-6881c9610e/docs/autoscale/v1/developer-guide/)

**Comment by manishtomar** [comment]
jenkins please run integration tests

**Comment by manishtomar** [comment]
@glyph review please.

File: otter/models/intents.py [hunk 1]
```diff
@@ -78,6 +78,25 @@ def perform_update_group_status(dispatcher, ugs_intent):
...
+    of group object
+    """
+    tenant_id = attr.ib()
+    group_id = attr.ib()
+    status = attr.ib()
+
+
+@deferred_performer
+def perform_update_scaling_group_status(log, store, dispatcher, intent):
+    """Perform an :obj:`UpdateScalingGroupStatus`."""
```
Comment (glyph) [inline otter/models/intents.py]: Can you add some `:param`s here?  https://txeffect.readthedocs.io/en/latest/ is not much help in deciphering the expected signature of a `@deferred_performer`, unfortunately.

File: otter/test/convergence/test_service.py [hunk 2]
```diff
@@ -857,7 +858,7 @@ def setUp(self):
         self.now = datetime(1970, 1, 1)
         self.waiting = Reference(pmap())
 
-    def get_seq(self, with_cache=True):
+    def get_seq(self, with_cache=True, upd_status_nogroup_err=False):
```
Comment (glyph) [inline otter/test/convergence/test_service.py]: That parameter's role is far from obvious to the casual reader: this needs a docstring :)

File: otter/models/intents.py [hunk 3]
```diff
@@ -78,6 +78,25 @@ def perform_update_group_status(dispatcher, ugs_intent):
 
 
 @attr.s
+class UpdateScalingGroupStatus(object):
```
Comment (glyph) [inline otter/models/intents.py]: This class name confused me at first.  The salient property here is not "scaling", as "group" and "scaling group" are synonyms, but rather the fact that one is *loaded already* and one isn't.  Do you think a name-change here to `LoadAndUpdateGroup` would make sense?  Given that there's an "and" in the name, is there any way to more gracefully compose the group-loading and existing group-updating effects?  (If not, that's fine, but if we could get away with fewer effects that need distinct performers, that would be great.)

**Comment by glyph** [comment]
All in all this one looks like a slam dunk to me - I'd like more docs in a few places (as noted in inline comments), and I think the naming is unduly confusing for new maintainers -  but overall it seems pretty clear what is going on.  After addressing please land.

File: otter/models/intents.py (hunk 3 above)
Comment (manishtomar) [inline otter/models/intents.py]: Perfect! Thanks for coming out with much better name! Will change it.

File: otter/models/intents.py (hunk 1 above)
Comment (manishtomar) [inline otter/models/intents.py]: Hmm. This is not good as lot of code has limited documentation. It leads me to believe that we need much more documentation for effect related code :-|

File: otter/test/convergence/test_service.py (hunk 2 above)
Comment (manishtomar) [inline otter/test/convergence/test_service.py]: Oh it needs more than a docstring. Its one of those messy tests that I would like to rewrite. However, I'll add a docstring to explain.

**Pull request merged**